ccswbs/gus
# Add customizable OG metatags for URL Preview Cards

# Summary of Changes

## Frontend

- Add `og:image` and `og:image:alt` variables to `seo.js`
- Add default values for `og:image` and `og:image:alt` to `ugconthub.js`
- Reference new defaults in `gatsby.config.js` metaConfig section
- Add Open Graph variables to `basic-page.js` and `program-page.js`
- If hero image present, use it for `og:image` and `og:image:alt`
- Change SEO component call in `basic-page.js` and `program-page.js` to send og values for description, image, and alt text
- Add `field_metatags` to Basic Page and Program schema
- Refactor hero image code on `program-page.js`
- Remove unneeded file `aplacetogrow.js`

## Backend

- Add metatag module
- Enable Open Graph fields for Basic Page, Program Page, Event, and News
- Add `field_metatags` to Basic Page and Program Page content types
- Configure metatag permissions so Author and Editor roles will only see the Open Graph description field

## Test Plan

Please note that you will need to use the browser inspector tool to view the values below and they will most likely appear just above the `</head>` closing tag:

- Go to https://tqtest.gatsbyjs.io/
- View page https://tqtest.gatsbyjs.io/studentexperience/faith-and-spirituality and confirm it has a custom `og:description`
- View page https://tqtest.gatsbyjs.io/land-acknowledgement and confirm it has custom values for `og:image`, `og:image:alt`, and `og:description`
- Visit https://tqtest.gatsbyjs.io/programs/test-program and confirm it has custom values for `og:image`, `og:image:alt`, and `og:description`
- Do the same for https://tqtest.gatsbyjs.io/programs/bachelor-of-arts-general-fully-online
- For https://tqtest.gatsbyjs.io/programs/bachelor-of-arts confirm that UoG default values are used instead since it has no hero image and no custom description

**Note:** https://api.devugconthub.uoguelph.dev/ is being used as the Drupal backend. There is no way to test the final URL preview cards until these changes are live (since both Twitter and Facebook's debugging tools refuse to recognize any Gatsby Cloud URLs)
